src/common/mfu_util.c

```diff
@@ -36,6 +36,111 @@ mfu_loglevel mfu_debug_level = MFU_LOG_ERR;
 /* default progress message timeout in seconds */
 int mfu_progress_timeout = 10;
 
+/***** DAOS utility functions ******/
+#ifdef DAOS_SUPPORT
+bool daos_uuid_valid(const uuid_t uuid)
+{
+	return uuid && !uuid_is_null(uuid);
+}
+
+/* Distribute process 0's pool or container handle to others. */
+void HandleDistribute(int rank, daos_handle_t *handle,
+                      daos_handle_t* poh, enum handleType type)
+{
+        d_iov_t global;
+        int        rc;
+
+        global.iov_buf = NULL;
+        global.iov_buf_len = 0;
+        global.iov_len = 0;
+
+        if (rank == 0) {
+                /* Get the global handle size. */
+                if (type == POOL_HANDLE)
+                        rc = daos_pool_local2global(*handle, &global);
+                else
+                        rc = daos_cont_local2global(*handle, &global);
+                if (rc != 0) {
+                    MFU_LOG(MFU_LOG_INFO, "Failed to get global handle size");
+                }
+        }
+
+        MPI_Bcast(&global.iov_buf_len, 1, MPI_UINT64_T, 0, MPI_COMM_WORLD);
+ 
+	global.iov_len = global.iov_buf_len;
+        global.iov_buf = malloc(global.iov_buf_len);
+        if (global.iov_buf == NULL)
+		MPI_Abort(MPI_COMM_WORLD, -1);
+
+        if (rank == 0) {
+                if (type == POOL_HANDLE)
+                        rc = daos_pool_local2global(*handle, &global);
+                else
+                        rc = daos_cont_local2global(*handle, &global);
+                if (rc != 0) {
+                    MFU_LOG(MFU_LOG_INFO, "Failed to create global handle");
+                }
+        }
+
+        MPI_Bcast(global.iov_buf, global.iov_buf_len, MPI_BYTE, 0, MPI_COMM_WORLD);
+
+        if (rank != 0) {
+                if (type == POOL_HANDLE)
+                        rc = daos_pool_global2local(global, handle);
+                else
+                        rc = daos_cont_global2local(*poh, global, handle);
+
+                if (rc != 0) {
+                    MFU_LOG(MFU_LOG_INFO, "Failed to get local handle");
+                }
+        }
+
+        free(global.iov_buf);
+}
+
+void daos_connect(int* rank, daos_handle_t* poh, daos_handle_t* coh,
```

Comment:
There are some extra/missing indentation on various lines in this function.

Reply:
@adammoody ok, fixed those.